### DISCUSSION TITLE
Batch file to open a set of books at specified page numbers

### DISCUSSION THREAD

**Discussion by Eagle20236** [sumatrapdfreader/sumatrapdf#4154] [Q&A] [answered]
I tried to write a batch file that would open a group of books at a specified page number in Sumatra. A typical line of my batch file includes:

"D:\EBooks\Jacobs - Quantum Mechanics 2nd Ed (2012).pdf" -page 20

This opens Sumatra and loads the book but it does not open it at the required page number. 
How can I amend the batch file so that it opens the book at the specified page?

**Comment by GitHubRulesOK** [comment]
unfortunately that FTA method is not used by Windows Cmdr. the order would need to be reversed

"...........\SumatraPDF.exe" -page %variable #%  "%filename%"

so something like 
`"....sumatrapdf" -page 20 "D:\EBooks\Jacobs - Quantum Mechanics 2nd Ed (2012).pdf"`

**Reply by Eagle20236** [reply]
Thanks for the suggestion. However that does not work either. It opens the book but not at the required page number.

**Reply by GitHubRulesOK** [reply]
ODD it should not be a problem so which version of exe ? and please copy paste exact CMD/BAT content (if lots of entries trim it down to a few)

**Reply by GitHubRulesOK** [reply]
Note the current pre-release can exhibit issues over what is **current state** (using historic stored page entry) thus it is often better to enforce reuse such as 

`"%programfiles%\sumatrapdf\sumatrapdf" -reuse-instance -page 2 "signed_doc2t.pdf"

**Comment by Eagle20236** [comment]
Thank you for a fast reply, using `-reuse-instance` solved the problem.

Another issues has arisen. My batch file contains two books (I have downloaded these from Project Gutenberg for this demonstration at `https://www.gutenberg.org/ebooks/search/?query=physics&submit_search=Go%21`)  as shown:

```
"D:\Program Executables\SumatraPDF\SumatraPDF-3.5.2-64.exe"  -reuse-instance -page 14 "D:\EBooks\EBook collection\Non-fiction\Maths\A Course of Pure Mathematics.pdf"

"D:\Program Executables\SumatraPDF\SumatraPDF-3.5.2-64.exe"  -reuse-instance -page 34 "D:\EBooks\EBook collection\Non-fiction\Maths\Calculus\Calculus made easy (Guttenburg).pdf"
```

This works well if Sumatra is already open. However, if I run the batch file and Sumatra is not open then Sumatra opens only the first book at the correct page. When I shut down Sumatra the batch file re-opens Sumatra and opens the second book at the correct page.

If I insert the following line at the top of the batch file then Sumatra opens when I run the batch file and none of the books in the batch file open:

`"D:\Program Executables\SumatraPDF\SumatraPDF-3.5.2-64.exe"`

When I close Sumatra the batch file reopens SUmatra with the first book at the required page and when I close Sumatra again the batch file re-opens Sumatra with the second book at the required page.

How can I amend the batch file so that if Sumatra is closed when I run the batch file all the books in the batch file will open at the correct page without me having to close Sumatra each time?

**Reply by GitHubRulesOK** [reply]
That is a challenge as its down to unknown states at the time so best solution that springs to mind is use "start" first
unsure how that will work on your system but try

```
set "SumatraPDF=D:\Program Executables\SumatraPDF\SumatraPDF-3.5.2-64.exe"
start "SumatraPDF" "%SumatraPDF%"
"%SumatraPDF%"  -reuse-instance -page 14 "D:\EBooks\EBook collection\Non-fiction\Maths\A Course of Pure Mathematics.pdf"
"%SumatraPDF%"  -reuse-instance -page 34 "D:\EBooks\EBook collection\Non-fiction\Maths\Calculus\Calculus made easy (Guttenburg).pdf"
```

**Reply by Eagle20236** [reply, marked as answer]
Thank you, that works on my computer.
